WordPress/gutenberg
# Migrate site editor to use new preferences package

## Description
Addresses some of https://github.com/WordPress/gutenberg/issues/31965

Migrates the site editor to use the new preferences package (introduced in https://github.com/WordPress/gutenberg/pull/38873). Previously this package had its own preferences implementation, but now it uses a common package, which should speed up the process of implementing new preferences.

The main goal here is that nothing is different from a user perspective.

## Testing Instructions
1. Delete your WP localstorage data
2. Using `trunk` open the site editor and dismiss the welcome guide
3. Toggle some of the preferences in the More Menu to something other than the default (e.g. set Top toolbar to active)
4. Checkout this branch and rebuild
5. Reload the site editor
6. The welcome guide should not show and the preferences set in step 3 should still be the same (the data migration worked)
7. Change a few more preferences
8. Reload
9. The preferences should still be the same from step 7 (preference changes are still persisted)

## Checklist:
